spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -197,6 +201,34 @@ private Duration toDuration(String time) {
     }
   }
 
+  class EnumConfParser<T extends Enum<T>> extends ConfParser<EnumConfParser<T>, T> {
+    private T defaultValue;
+    private final Function<String, T> toEnum;
```

Comment:
Minor: We usually have final vars before mutable. Can we switch the order of these two?